torch/testing/_asserts.py

```diff
@@ -219,12 +205,19 @@ def _trace_mismatches(actual: Tensor, expected: Tensor, mismatches: Tensor) -> _
     )
 
 
-def _check_values_equal(actual: Tensor, expected: Tensor) -> Optional[AssertionError]:
+def _check_values_equal(
+    actual: Tensor,
+    expected: Tensor,
+    *,
+    msg: Optional[Union[str, Callable[[Any, Any, SimpleNamespace], str]]] = None,
+) -> Optional[AssertionError]:
     """Checks if the values of two tensors are bitwise equal.
 
     Args:
         actual (Tensor): Actual tensor.
         expected (Tensor): Expected tensor.
+        msg (Optional[Union[str, Callable[[Any, Any, SimpleNamespace], str]]]): Optional error message. Can be passed
+            as callable in which case it will be called with the inputs and the result of :func:`_trace_mismatches`.
```

Comment:
Rather than a callable, could this be a prefix to the standard message? Are there cases in the test suite that require this to be a callable?

Reply:
Not yet, but the idea for custom messages arose from the need to make the error messages of a binary op test more clear. Have a look at the example I gave in the initial comment.